frontend/src/lib/k8s/cluster.ts

```diff
@@ -36,6 +36,29 @@ export interface Cluster {
   [propName: string]: any;
 }
 
+export class BaseKubeObject {
+  static getBaseObject(): KubeObjectInterface {
+    return {
+      apiVersion: 'v1',
+      kind: '',
+      metadata: {
+        name: '',
+        namespace: '',
+        annotations: {},
+        labels: {},
+        creationTimestamp: '',
+        uid: '',
+      },
+    };
+  }
+}
+
+export interface ClassWithBaseObject<T> {
+  new (...args: any[]): T;
+  kind?: string;
+  getBaseObject?(): KubeObjectInterface;
+}
+
```

Comment:
This shouldn't be needed. The getBaseObject should be added to the KubeObject class in cluster.ts .

Reply:
Moved the function and removed the unneeded changes here